### PR TITLE
[#709] Added Submission.form_url

### PR DESCRIPTION
**Pull request by Bartvaderkin**
Added Submission.form_url, and dropped SubmissionPayment.form_url and GlobalConfiguration.cancel_appointment_page

This makes it required to send `formUrl` when starting a Submission.

### CODE REVIEW

File: src/openforms/appointments/tests/test_endpoints.py [hunk 1]
```diff
@@ -42,7 +39,7 @@ def test_good_token_and_submission_redirect_and_add_submission_to_session(self):
             response = self.client.get(endpoint)
 
         expected_redirect_url = (
-            furl("http://maykinmedia.nl/afspraak-annuleren")
+            furl("http://maykinmedia.nl/myform#afspraak-annuleren")
```
Comment (sergei-maertens) [inline src/openforms/appointments/tests/test_endpoints.py:42]:
```suggestion
            furl("http://maykinmedia.nl/myform/afspraak-annuleren")
```
we don't do client-side routing based on hash-fragments, but actual URLs :)

File: src/openforms/appointments/views.py [hunk 2]
```diff
@@ -33,20 +33,12 @@ def get_redirect_url(self, submission_uuid: uuid, token: str, *args, **kwargs):
...
-            furl(config.cancel_appointment_page)
-            .add(
-                {
-                    "time": submission.appointment_info.start_time.isoformat(),
-                    "submission_uuid": submission.uuid,
-                }
-            )
-            .url
+        f = furl(submission.form_url)
+        f.fragment.path.segments = ["afspraak-annuleren"]
```
Comment (sergei-maertens) [inline src/openforms/appointments/views.py:37]: should not be fragment, but be added to the path

File: src/openforms/conf/base.py [hunk 3]
```diff
@@ -204,7 +204,7 @@
     "django.contrib.messages.middleware.MessageMiddleware",
     "django.middleware.clickjacking.XFrameOptionsMiddleware",
     "axes.middleware.AxesMiddleware",
-    "csp.contrib.rate_limiting.RateLimitedCSPMiddleware",
+    # "csp.contrib.rate_limiting.RateLimitedCSPMiddleware",
```
Comment (sergei-maertens) [inline src/openforms/conf/base.py:207]: accidental commit?

File: src/openforms/payments/views.py [hunk 4]
```diff
@@ -115,23 +115,22 @@ def post(self, request, uuid: str, plugin_id: str):
...
+        # form_url = request.query_params.get("next")
+        # if not form_url:
+        #     raise ParseError(detail="missing 'next' parameter")
+        #
+        # if not allow_redirect_url(form_url):
+        #     logger.warning(
+        #         "blocked payment start with non-allowed redirect to '%(form_url)s'",
+        #         {"form_url": form_url},
+        #     )
+        #     raise ParseError(detail="redirect not allowed")
```
Comment (sergei-maertens) [inline src/openforms/payments/views.py:127]:
```suggestion
```

File: src/openforms/submissions/models.py [hunk 5]
```diff
@@ -134,6 +134,14 @@ class Submission(models.Model):
         blank=True,
         validators=(validate_kvk,),
     )
+    form_url = models.URLField(
+        _("Form URL"),
```
Comment (sergei-maertens) [inline src/openforms/submissions/models.py:138]:
```suggestion
        _("form URL"),
```

**Review by sergei-maertens — CHANGES_REQUESTED** [review]
Couple minor changes, looks good otherwise

File: src/openforms/conf/base.py (hunk 3 above)
Comment (Bartvaderkin) [reply src/openforms/conf/base.py:207]: accidental commit, but not accidental edit: this generates a ton of CSP-report requests (on status calls, json logic etc).

**Review by sergei-maertens — APPROVED** [review]

**Pull request merged**